src/net/connection/virtual_connection.rs

```diff
@@ -56,87 +60,345 @@ impl VirtualConnection {
     }
 
     /// This pre-process the given buffer to be send over the network.
-    /// 1. It will append the right header.
-    /// 2. It will perform some actions related to how the packet should be delivered.
-    pub fn process_outgoing(
+    ///
+    /// # Algorithm
+    /// 1. Write `StandardHeader`
```

Comment:
As nice as these are today, I really dislike having the algorithm description as a comment. Inevitably, we're going to make changes to the code and forget to update the comment which will leave the comment misleading/useless. Let the code speak for itself please :).

Reply:
That is part of the problem, I agree that the code should speak for its self. But I walk some problems here and I like your thoughts on it. 

I was thinking about moving all the processing logic into different types and have some extensible way to handle lot's of different parsing scenario's, but I came to the conclusion that putting the processing logic in a single function was making

There where two reasons why I did this:
1. We don't have a lot of handling scenarios and thus we don't need a very extensible way of handling packets as described in #143.  
2.  It is much easier to handle an incoming or outgoing packet if the processing logic is at one place. 

One downside that comes from processing packets in one function is that the function will be a bit bigger than normal functions should be. One can state that they only might be around 7 lines because that is the attention span of the reader. So I thought that it would be helpful for the user to add a short bullet list to help the reader to understand what this does. 

Three questions and notices:

1. Do you think we should break this down and make this packet processing more extensible?
2. Do you think this code does not explain itself? 
3.   Do I need to remove those comments; I can do that but I don't really see the purpose of that since I think it is valuable for the reader. But sure if you want to I can remove it. 

I was already thinking of braking this more down, and see what more I can do to make it a bit smaller but keep the easy maintainability of the code in a next PR because this PR would become to big.

Reply:
7 lines!? That's extremely small. While methods/functions should err on the smaller side, having a function that is 50-100 lines long is common and reasonable in my opinion. The heuristic I like to use is whether or not I can see the entire function's implementation without scrolling. Any bigger and I consider breaking it up a little. There is such a thing as having too much abstraction. It makes the code hard to read in the other direction because you're constantly jumping between methods/modules/files to get a full picture of what's going on. So to specifically answer your questions,
1. I think this is perfectly fine
2. I think the code explains itself pretty well.
3. I would remove the comments explaining the algorithm because the code itself is explaining the algorithm in a more accurate way.

Reply:
Alright, nice to hear, let me remove them.